### PR TITLE
Log `searchResults:ownershipMailto:clicked` and other clicks

### PR DESCRIPTION
**Pull request by cbart**
Part of #47062

FYI @limitedmage I've changed the linking conditional.

```diff
-       if (result.type === 'person' && !result.user) {
+       const validUrlPrefixes = ['/teams/', '/users/', 'mailto:']
+       if (!validUrlPrefixes.some(prefix => url.startsWith(prefix))) {
            // This is not a real URL, remove it.
            return ''
        }
```

The former one would only link resoled users, whereas now it links resolved users, teams, and mailto:'s. Is that OK?

Events added (each is a click on search result):

- `searchResults:ownershipMailto:clicked` - if it's a mailto link
- `searchResults:ownershipUsers:clicked` - if it's a link to a resolved user
- `searchResults:ownershipTeams:clicked` - if it's a link to a resolved team

## Test plan

https://www.loom.com/share/ce201622d87842e29346c82c966b5aa6

## App preview:

- [Web](https://sg-web-cb-own-amplitude.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | 0.00% (+0.29 kb) | 0.00% (+0.29 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/cb/own-amplitude-searchOwnershipMailto/compare-report.html;tab=live_object) for a full comparison between the commits 5afcb24 and 2bb3963 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: client/branded/src/search-ui/components/OwnerSearchResult.tsx [hunk 1]
```diff
@@ -40,7 +42,8 @@ export const OwnerSearchResult: React.FunctionComponent<PersonSearchResultProps>
 
     const url = useMemo(() => {
         const url = getOwnerMatchUrl(result)
-        if (result.type === 'person' && !result.user) {
+        const validUrlPrefixes = ['/teams/', '/users/', 'mailto:']
+        if (!validUrlPrefixes.some(prefix => url.startsWith(prefix))) {
```
Comment (limitedmage) [inline client/branded/src/search-ui/components/OwnerSearchResult.tsx:46]: I like this much better than my hacky stuff 👍

File: client/branded/src/search-ui/components/OwnerSearchResult.tsx [hunk 2]
```diff
@@ -68,7 +71,21 @@ export const OwnerSearchResult: React.FunctionComponent<PersonSearchResultProps>
...
+                        if (url.startsWith('mailto:')) {
+                            telemetryService.log('searchResults:ownershipMailto:clicked')
+                        }
+                        if (url.startsWith('/users/')) {
+                            telemetryService.log('searchResults:ownershipUsers:clicked')
+                        }
+                        if (url.startsWith('/teams/')) {
+                            telemetryService.log('searchResults:ownershipTeams:clicked')
+                        }
+                    }}
```
Comment (limitedmage) [inline client/branded/src/search-ui/components/OwnerSearchResult.tsx:87]: nit: this is a bit long for inline, I recommend moving it out

**Review by limitedmage — APPROVED** [review]

File: client/branded/src/search-ui/components/OwnerSearchResult.tsx (hunk 2 above)
Comment (cbart) [reply client/branded/src/search-ui/components/OwnerSearchResult.tsx]: Extracted!

File: client/branded/src/search-ui/components/OwnerSearchResult.tsx (hunk 1 above)
Comment (cbart) [reply client/branded/src/search-ui/components/OwnerSearchResult.tsx:46]: 🎉

**Review by eseliger — APPROVED** [review]

**Pull request merged**